src/components/Discover/TrendingTokens.tsx

```diff
@@ -508,12 +523,15 @@ function NoResults() {
   );
 }
 
-function NetworkFilter() {
-  const chainId = useTrendingTokensStore(state => state.chainId);
-  const selected = useSharedValue<ChainId | undefined>(chainId);
+function NetworkFilter({ selectedChainId }: { selectedChainId: SharedValue<ChainId | undefined> }) {
+  const { chainId, category } = useTrendingTokensStore(state => ({
+    chainId: state.chainId,
+    category: state.category,
+  }));
```

Comment:
Should split or use `shallowEqual`, otherwise this will be recreated each render